### PR TITLE
update CI images to 20230824

### PR DESCRIPTION
**Pull request by Luap99**
includes the en_US.UTF-8 locale

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by Luap99** [comment]
weird
```
/usr/share/automation/lib/utils.sh: line 107: nc: command not found
```

Did we remove nc from the images? cc @cevich

**Comment by cevich** [comment]
Not intentionally, and maybe an interesting note is this didn't fail in the slightly-older `20230816t191118z-f38f37d13`.  For Debian there is [some nc/ncat Debian-specific magic](https://github.com/containers/automation_images/blob/main/cache_images/debian_packaging.sh#L133-L134) @edsantiago helped me put in.  Maybe the image build logs have some clues...

**Comment by cevich** [comment]
...oof, it get's more strange.  [The build log](https://cirrus-ci.com/task/5374869337014272?logs=main#L1509) shows the magic happening:

```
 debian: update-alternatives: using /usr/bin/ncat to provide /bin/nc (nc) in auto mode
```

Maybe we need a `nc` symlink also?  Maybe there's a bug in SID that caused this?

**Comment by Luap99** [comment]
Well it fails on both debian and fedora so this is really odd that both are breaking at the same time.

**Comment by cevich** [comment]
Oh good point, I didn't notice the Fedora failures.  Hrmmmm, that's concerning.  Let's see what get_ci_vm.sh says...

**Comment by edsantiago** [comment]
Nothing is installed:
```
[root@cirrus-task-4949830783467520 .cirrus]# rpm -q pandoc
package pandoc is not installed
[root@cirrus-task-4949830783467520 .cirrus]# rpm -q libtool
package libtool is not installed
```
("pandoc" and "libtool" are taken from the list in `cache_images/*.sh`)

Of course by "nothing" I don't mean it literally, only that things-that-I-expect-to-be-installed are not-installed. `nc` seems to be only the first one detected.

**Comment by Luap99** [comment]
Ok I assume the images are just broken then??? @cevich can you respin a new build.

File: .cirrus.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ env:
     DEBIAN_NAME: "debian-13"
 
     # Image identifiers
-    IMAGE_SUFFIX: "c20230816t191118z-f38f37d13"
+    IMAGE_SUFFIX: "b20230821t153739z-f38f37d13"
```
Comment (edsantiago) [inline .cirrus.yml:36]: OH! This should be **c**, not b, I think? @cevich?

Comment (Luap99) [reply .cirrus.yml:36]: oh yes, whoops I copy-pasted the wrong ID

Comment (cevich) [reply .cirrus.yml]: Opened https://github.com/containers/automation_images/pull/296

**Comment by edsantiago** [comment]
Pasta is broken:
* OLD:  passt-0^2023**0625**.g32660ce-1.fc38-x86_64  - works
* NEW: passt-0^2023**0818**.g0af928e-1.fc38-x86_64 - fails

@sbrivio-rh it looks like passt **0818** is broken, could you PTAL at the failing tests, and fix either pasta or the tests?

**Comment by sbrivio-rh** [comment]
> Pasta is broken:
> 
>     * OLD:  passt-0^2023**0625**.g32660ce-1.fc38-x86_64  - works
>     * NEW: passt-0^2023**0818**.g0af928e-1.fc38-x86_64 - fails
> 
> @sbrivio-rh it looks like passt **0818** is broken, could you PTAL at the failing tests, and fix either pasta or the tests?

Oops, yes, sorry, I guess pasta is broken, we get [EINVAL while copying IPv4 routes](https://cirrus-ci.com/task/4572603738750976?logs=main#L553). I'm having a look now, and tagging @dgibson in case I fall asl

**Comment by dgibson** [comment]
So, the change to report (most) errors on netlink operations made the failure here more evident, but it already existed.  Previously rather than outright failing, we would simply have omitted copying certain routes from the container.  Depending what those routes are the container might still operate more or less ok, or it might completely.

The reason certain routes are seeing `EINVAL` -  at least in the cases I've reproduced is that those routes have an `RTA_SRC`, limiting the route to a specific source address on the host.  When we use the `-a` option, as the failing examples do, the guest no longer has that address, and so it's invalid to add a route with that address as a source.

**Comment by dgibson** [comment]
Correction: the issue described in the previous comment is only causing one of the podman failures.  There's another bug causing some others, which @sbrivio-rh is looking into.

**Comment by cevich** [comment]
So for this PR (which is not networking related?), it sounds like the "file-issue + skip tests" strategy is most appropriate.  I'm going to merge https://github.com/containers/automation_images/pull/297 as-is for now.  But if some quick change is needed in the CI VM images, I'm happy to assist with that.

**Comment by sbrivio-rh** [comment]
> So for this PR (which is not networking related?), it sounds like the "file-issue + skip tests" strategy is most appropriate. I'm going to merge [containers/automation_images#297](https://github.com/containers/automation_images/pull/297) as-is for now. But if some quick change is needed in the CI VM images, I'm happy to assist with that.

As you wish -- I'm preparing new packages for Fedora and Debian at the moment, but it might take a while and if you have a way out...

By the way, fixes are:

https://passt.top/passt/commit/?id=5e4f7b92b0b0bf4724c505fa95fcae1526a8f88b
https://passt.top/passt/commit/?id=a7e4bfb857cb5d0e111ab74b6ace47eea15d2078

**Comment by Luap99** [comment]
There is nothing urgent in this PR. This is a simple cleanup of my previous work, we can wait until packages land in stable then rebuild images.
Although I will be on PTO the next two weeks so would be much appreciated if someone else could push the image updates if that happens in the meantime.

**Comment by cevich** [comment]
> There is nothing urgent in this PR. This is a simple cleanup of my previous work, we can wait until packages land in stable then rebuild images.

Okay SGTM.  @sbrivio-rh do you have a way to monitor those packages?  Once they land in SID and Fedora updates-testing, I'm happy to roll new CI VM images and get them updated for Paul.

**Comment by sbrivio-rh** [comment]
> > There is nothing urgent in this PR. This is a simple cleanup of my previous work, we can wait until packages land in stable then rebuild images.
> 
> Okay SGTM. @sbrivio-rh do you have a way to monitor those packages? Once they land in SID and Fedora updates-testing [...]

I'm the maintainer for both -- I'll post here when they do.

**Comment by cevich** [comment]
Fantastic, that will help a lot, thanks.

**Comment by sbrivio-rh** [comment]
Fedora 38 [update](https://bodhi.fedoraproject.org/updates/FEDORA-2023-3a7b730bc9) should reach testing in a bit, Debian Unstable [package](https://qa.debian.org/cgi-bin/vcswatch?package=passt) available for upload. I'll tell you anyway once they reach the repositories.

**Comment by TomSweeneyRedHat** [comment]
changes LGTM, but the tests still need to be muscled over the line.

**Comment by cevich** [comment]
Great thanks, I'll start building fresh CI VM images.

**Comment by Luap99** [comment]
@cevich I used the image ID from https://github.com/containers/automation_images/pull/298

@sbrivio-rh looks like fedora is working again but there is still a pasta issue on debian:

```
[+1329s] # [13:49:24.955653397] Error: pasta failed with exit code 1:
[+1329s] # Can't run AVX2 build, using non-AVX2 version: No such file or directory
[+1329s] # No external routable interface for IPv6
[+1329s] # Couldn't get any nameserver address
[+1329s] # Template interface: ens4 (IPv4)
[+1329s] # Namespace interface: myname
[+1329s] # MAC:
[+1329s] #     host: 42:01:0a:80:00:ed
[+1329s] # DHCP:
[+1329s] #     assign: 10.128.0.237
[+1329s] #     mask: 255.255.255.255
[+1329s] #     router: 10.128.0.1
[+1329s] # mount /: Permission denied
[+1329s] # Failed to sandbox process, exiting
```

**Comment by cevich** [comment]
Paul! You're suppose to be on PTO!

Moving this issue to https://github.com/containers/podman/pull/19751

**Comment by Luap99** [comment]
I am on PTO the next two weeks, so I still have an hour until PTO/weekend starts for me.

**Comment by cevich** [comment]
No worries, I was just teasing.  Enjoy your PTO :smile:

**Pull request closed without merging**